### PR TITLE
Feature/us5243 inline giving prototype updates

### PR DESCRIPTION
**Pull request by zarahzachz**
Update prototype to include payment flow

If a transaction is a payment/non-contribution, 
then display a new payment screen that should include the following fields:
- Title (name of the product or invoice)
- Maximum amount (total amount for the invoice)
- Minimum payment (amount due immediately, the deposit amount)

There are scenarios where the maximum payment == the minimum payment (e.g. has to be paid in full up front), then make UI update to account for this 

<-----Payments----->
Add Account Holder Name field to Bank Account

<---Confirmation Screen Updates--->
Update the confirmation screen to have universal labels (that work for both non-contribution and contribution transactions)

Reconsider whether or not we should link back to the steps from the confirmation screen, there is a significantly amount of complexity there

Do not display ability to edit email for authenticated users; only change email for guest givers

on embedint, email field was not displaying

### CODE REVIEW

**Comment by beckysingson** [comment]
@zarahzachz The confirmation screen for the is not displaying email address for either flow. It should show here with an edit button for people giving as guest (no edit for logged in users). Otherwise this looks great!

**Comment by zarahzachz** [comment]
Email field will now appear in both payment/gift flows regardless of user type. Updated the change email link to appear only when a guest user is signed in.

File: src/app/prototype/prototype-payment-amount/prototype-payment-amount.component.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    if (!isNaN(newValue)) {
+      delete(this.selectedAmount);
+      this.setAmount(newValue);
+    }
+  }
+
+  onSelectAmount(event, newValue) {
+    delete(this.customAmount);
+    this.setAmount(newValue);
+    console.log(newValue);
```
Comment (samueljmello) [inline src/app/prototype/prototype-payment-amount/prototype-payment-amount.component.ts]: @zarahzachz Should remove this console log

**Comment by zarahzachz** [comment]
## Post 10/25/16 Meeting with Falcons
- Remove product title from page 1, but keep on summary page
- Replace 'Product' field label on summary page with 'To' field label (will have the same label for both payment and gift flows)
- Remove 'Give as Guest' option on authentication screen for payment flow
- Make text in final submit button condition: Pay $X for payment flow and Give $X for gift flow

**Comment by beckysingson** [comment]
@zarahzachz this looks great! Thanks for the updates!

File: src/app/prototype/prototype-switch/prototype-switch.component.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+})
+export class PrototypeSwitchComponent {
+
+  constructor(@Inject(PrototypeStore) private store: any,
+              private gift: PrototypeGiftService) {
+
+    this.gift.flow_type = this.gift.flow_type == 'payment' ? 'gift' : 'payment';
+    this.gift.reset();
+
+    console.log(this.gift.flow_type);
```
Comment (samueljmello) [inline src/app/prototype/prototype-switch/prototype-switch.component.ts]: Please remove console.log

File: src/app/prototype/prototype-email/prototype-email.component.html [hunk 3]
```diff
@@ -2,9 +2,9 @@
   <h2>Change Email</h2>
 </header>
 
-<div class="alert alert-warning">
-  <strong>NOTE:</strong> Any change to your email here will be reflected throughout crossroads.net, including your login credentials. 
-</div>
+<!-- <div class="alert alert-warning">
+  <strong>NOTE:</strong> Any change to your email here will be reflected throughout crossroads.net, including your login credentials.
+</div> -->
```
Comment (samueljmello) [inline src/app/prototype/prototype-email/prototype-email.component.html]: Please remove commented-out code

**Pull request merged**